### DISCUSSION TITLE
[Idea] Add support for Imagemagick

### DISCUSSION THREAD

**Discussion by eternoendless** [PrestaShop/PrestaShop#32254] [Ideas]
#### Is your feature request related to a problem?

The GD library is used in PrestaShop to build thumbnails. It would be nice to support Imagemagick as well.

#### Describe the solution you'd like

PrestaShop uses Imagemagick if available, and falls back to GD if not.

#### Alternatives you've considered

N/A

#### Additional context

See [this issue](https://github.com/PrestaShop/PrestaShop/issues/20900)

[a comment by an automated account is not shown]

**Comment by matks** [comment]
Related topics:
- https://github.com/PrestaShop/PrestaShop/issues/10600
- https://github.com/PrestaShop/PrestaShop/pull/15631
- https://github.com/PrestaShop/PrestaShop/issues/18518

**Comment by Hlavtox** [comment]
@matks @eternoendless Randomly stepped on this topic on processwire forums. 
See: https://processwire.com/blog/posts/processwire-3.0.10-expands-image-resize-options/#gd-vs-imagemagick-examples

Imagick for the win 🚀

**Comment by hibatallahAouadni** [comment]
Hello @eternoendless 

Thanks for the suggestion, the Product Team will take a look.
Please stay tuned !

Ping @PrestaShop/product-team , what do you think?

**Comment by rzuf666** [comment]
A very important aspect is also the handling of ICC color profiles for photos. GD can't do it, ImageMagic can.

Currently, browsers have no problems handling these profiles. However, the big problem now is that when a product photo containing a color profile is added to Prestashop, it is removed by the GD library. Due to this photo, it loses its color saturation and looks completely different than the seller planned.

This is very important for sales because customers only have to see the product are photos, thanks to which they decide to buy. This is of great importance, for example in the fashion industry, where each photo is specially taken in the studio. Even at the stage of taking a photo, professional photographic equipment has predefined color profiles for the best color reproduction. Then such photos are processed by graphic designers before being placed in the store.

Support for ICC color profiles is very important in e-commerce today.

I'm joining this suggestion, ImageMagic supports ICC colors so it would be great if Prestashop could handle it natively. There are modules for that, but I think that mixing the thumbnails with modules is not the best idea.

**Comment by Peter-Liska** [comment]
Here is also a discussion and ideas on this issue:

[Imagick prestashop forum](https://www.prestashop.com/forums/topic/447377-improving-the-image-quality/)

**Comment by DracoBluee** [comment]
How ImageMagick Optimization Image?

**Comment by Progi1984** [comment]
I work on it.

[EDIT] And finally not, no enough spare time.

**Comment by DracoBluee** [comment]
Great

**Comment by madsoliver** [comment]
What is missing to finalize this?
Seems like it's  almost finished?

**Comment by Tanguy-SALMON** [comment]
Up

**Comment by toby11** [comment]
has this been completed ?
